crypto/crypto/src/hash.rs

```diff
@@ -103,15 +103,21 @@ use anyhow::{ensure, Error, Result};
 use bytes::Bytes;
 use libra_nibble::Nibble;
 use mirai_annotations::*;
-use once_cell::sync::Lazy;
+use once_cell::sync::{Lazy, OnceCell};
 #[cfg(any(test, feature = "fuzzing"))]
 use proptest_derive::Arbitrary;
 use rand::{rngs::OsRng, Rng};
 use serde::{de, ser};
 use std::{self, convert::AsRef, fmt, str::FromStr};
 use tiny_keccak::{Hasher, Sha3};
 
-pub(crate) const LIBRA_HASH_PREFIX: &[u8] = b"LIBRA::";
+/// A prefix used to begin the salt of every libra hashable structure. The salt
```

Comment:
IHMO long explanations like this get out of sync too easily and are better grouped in one place.